django/db/models/functions/comparison.py

```diff
@@ -143,6 +143,74 @@ def as_sqlite(self, compiler, connection, **extra_context):
         return super().as_sqlite(compiler, connection, function="MAX", **extra_context)
 
 
+class JSONArray(Func):
+    function = "JSON_ARRAY"
+    output_field = JSONField()
```

Comment:
Before we add this function, should we assess whether we do want this function in `db.models.functions.comparison`? I know that the existing `JSONObject` function is here, so naturally it does make sense to just put `JSONArray` here as well.

However, given your example use case of `JSONArray`, it seems these functions are more likely to be used for building JSON objects and arrays directly in the database to be returned to the app-level code, and not necessarily to be used for comparison with another database value. (JSON object comparison seems rarely useful 🤔)

Also, consider another upcoming feature #18758, as well as the potential of the other JSON functions you mentioned in the ticket. I think we should create a new module for JSON-related functions, e.g. in `django.db.models.functions.json`, similar to how we have `django.db.models.fields.json` (but they should be exported from the higher-level module, i.e. `django.db.models.functions` as they are now).

If this is agreed, we should move `JSONObject` to that module as well (in a separate PR). It should be safe (I think), since we only document importing from `django.db.models.functions`.

Reply:
I would need some consensus on creating a new module.

It makes sense to me, but I'd like to hear other opinions.

Reply:
I like this direction :+1: I think we should do it
@charettes @felixxm what do you think?

Reply:
> @charettes @felixxm what do you think?

Agreed

Reply:
Makes sense to me as well.

Reply:
Thanks all! I've opened #18939 for this.

Reply:
I've rebased to move the class to the new module, as well as reorganizing the docs so it's in the "JSON" section with "JSONObject." I tried to keep things alphabetical, more or less.

Haven't made the requested change to the logic yet.